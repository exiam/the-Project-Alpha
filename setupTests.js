const Adapter = require('enzyme-adapter-react-16')

require('enzyme').configure({ adapter: new Adapter() })

require('dotenv').config({ path: '.env' })
