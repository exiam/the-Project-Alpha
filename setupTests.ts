import { configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
export function setup() {
  configure({ adapter: new Adapter() })
}
