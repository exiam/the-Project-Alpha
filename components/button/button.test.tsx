import * as React from 'react'
import Button from './button.component'
import { setup } from '../../setupTests'
setup()
import { render, shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'
describe('Button component', () => {
  describe('rendering', () => {
    it('should render without error', () => {
      render(<Button></Button>)
    })

    it('should render the text inside', () => {
      expect(render(<Button>text inside</Button>).text()).toBe('text inside')
    })
  })
  describe('interractions', () => {
    it('should call the onclick function when clicked', () => {
      const mockClick = jest.fn()
      const wrapper = shallow(<Button onClick={mockClick} />)
      wrapper.simulate('click')
      expect(mockClick).toHaveBeenCalled()
    })
  })
  describe('snapshot', () => {
    describe('primary', () => {
      it('should match with the primary snapshot', () => {
        expect(
          toJSON(shallow(<Button primary>text inside</Button>))
        ).toMatchSnapshot()
      })
      it('should match with the primary large snapshot', () => {
        expect(
          toJSON(
            shallow(
              <Button primary large>
                text inside
              </Button>
            )
          )
        ).toMatchSnapshot()
      })
    })
    describe('secondary', () => {
      it('should match with the secondary snapshot', () => {
        expect(toJSON(shallow(<Button>text inside</Button>))).toMatchSnapshot()
      })
      it('should match with the secondary large snapshot', () => {
        expect(
          toJSON(shallow(<Button large>text inside</Button>))
        ).toMatchSnapshot()
      })
    })
  })
})
