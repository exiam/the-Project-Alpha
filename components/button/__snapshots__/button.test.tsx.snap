// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component snapshot primary should match with the primary large snapshot 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "buttoncomponent__Button-sjtcji-0",
        "isStatic": false,
        "lastClassName": "elKMYf",
        "rules": Array [
          "background:",
          [Function],
          ";border:",
          [Function],
          " solid 3px;border-radius:5px;padding:7.5px 15px;color:",
          [Function],
          ";width:",
          [Function],
          "px;margin:6px;font-family:'Raleway',sans-serif;font-size:16px;font-weight:bold;transition-duration:0.3s;box-shadow:none;&:hover,&:focus{background:",
          [Function],
          ";color:white;box-shadow:3.5px 3.5px ",
          [Function],
          ";}&:active{background:",
          [Function],
          ";color:white;box-shadow:5.5px 5.5px ",
          [Function],
          ";}",
        ],
      },
      "displayName": "buttoncomponent__Button",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "buttoncomponent__Button-sjtcji-0",
      "target": "button",
      "toString": [Function],
      "usesTheme": false,
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  large={true}
  primary={true}
>
  text inside
</StyledComponent>
`;

exports[`Button component snapshot primary should match with the primary snapshot 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "buttoncomponent__Button-sjtcji-0",
        "isStatic": false,
        "lastClassName": "elKMYf",
        "rules": Array [
          "background:",
          [Function],
          ";border:",
          [Function],
          " solid 3px;border-radius:5px;padding:7.5px 15px;color:",
          [Function],
          ";width:",
          [Function],
          "px;margin:6px;font-family:'Raleway',sans-serif;font-size:16px;font-weight:bold;transition-duration:0.3s;box-shadow:none;&:hover,&:focus{background:",
          [Function],
          ";color:white;box-shadow:3.5px 3.5px ",
          [Function],
          ";}&:active{background:",
          [Function],
          ";color:white;box-shadow:5.5px 5.5px ",
          [Function],
          ";}",
        ],
      },
      "displayName": "buttoncomponent__Button",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "buttoncomponent__Button-sjtcji-0",
      "target": "button",
      "toString": [Function],
      "usesTheme": false,
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  primary={true}
>
  text inside
</StyledComponent>
`;

exports[`Button component snapshot secondary should match with the secondary large snapshot 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "buttoncomponent__Button-sjtcji-0",
        "isStatic": false,
        "lastClassName": "elKMYf",
        "rules": Array [
          "background:",
          [Function],
          ";border:",
          [Function],
          " solid 3px;border-radius:5px;padding:7.5px 15px;color:",
          [Function],
          ";width:",
          [Function],
          "px;margin:6px;font-family:'Raleway',sans-serif;font-size:16px;font-weight:bold;transition-duration:0.3s;box-shadow:none;&:hover,&:focus{background:",
          [Function],
          ";color:white;box-shadow:3.5px 3.5px ",
          [Function],
          ";}&:active{background:",
          [Function],
          ";color:white;box-shadow:5.5px 5.5px ",
          [Function],
          ";}",
        ],
      },
      "displayName": "buttoncomponent__Button",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "buttoncomponent__Button-sjtcji-0",
      "target": "button",
      "toString": [Function],
      "usesTheme": false,
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  large={true}
>
  text inside
</StyledComponent>
`;

exports[`Button component snapshot secondary should match with the secondary snapshot 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "buttoncomponent__Button-sjtcji-0",
        "isStatic": false,
        "lastClassName": "elKMYf",
        "rules": Array [
          "background:",
          [Function],
          ";border:",
          [Function],
          " solid 3px;border-radius:5px;padding:7.5px 15px;color:",
          [Function],
          ";width:",
          [Function],
          "px;margin:6px;font-family:'Raleway',sans-serif;font-size:16px;font-weight:bold;transition-duration:0.3s;box-shadow:none;&:hover,&:focus{background:",
          [Function],
          ";color:white;box-shadow:3.5px 3.5px ",
          [Function],
          ";}&:active{background:",
          [Function],
          ";color:white;box-shadow:5.5px 5.5px ",
          [Function],
          ";}",
        ],
      },
      "displayName": "buttoncomponent__Button",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "buttoncomponent__Button-sjtcji-0",
      "target": "button",
      "toString": [Function],
      "usesTheme": false,
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
>
  text inside
</StyledComponent>
`;
